### DISCUSSION TITLE
A couple mistakes in the wiki and a question

### DISCUSSION THREAD

**Discussion by gatisnolv** [ExistentialAudio/BlackHole#683] [General]
1. The https://github.com/ExistentialAudio/BlackHole/wiki/Multi-Output-Device page in point 4, it says "It is highly recommended you enable Drift Correction for all devices except the **Clock Source device**." It should say "Master Device" instead, which is the term used for Multi-Output Devices in the UI.

2. The https://github.com/ExistentialAudio/BlackHole/wiki/Aggregate-Device page in point 4 says "In the new Aggregate device you just created select the audio devices that you want to **output audio**". I think this is incorrect, and should say something like "be used as audio inputs" or "aggregate".

3. Related to my 2. point, could you clear up my understanding about this? Would it be fair to interpret the meaning of what Apple has called "Aggregate Device" as a "Multi-**Input** Device"? If so, what is the point of adding both the microphone (technically an input device) and also the speakers (technically and output device) to such an Aggregate Device? For example, there is no sound coming from the speakers if I add built-in speakers to such an aggregate device and select is as the sound output device of the system.

**Comment by devinroth** [comment]
On Nov 23, 2022, 5:03 AM -0800, gatisnolv ***@***.***>, wrote:

1. The https://github.com/ExistentialAudio/BlackHole/wiki/Multi-Output-Device page in point 4, it says "It is highly recommended you enable Drift Correction for all devices except the Clock Source device." It should say "Master Device" instead, which is the term used for Multi-Output Devices in the UI.

Apple updated the wording. I’ll change it.

1.
2. The https://github.com/ExistentialAudio/BlackHole/wiki/Aggregate-Device page in point 4 says "In the new Aggregate device you just created select the audio devices that you want to output audio". I think this is incorrect, and should say something like "be used as audio inputs" or "aggregate".

it can be used as both input or output.

2.
3. Related to my 2. point, could you clear up my understanding about this? Would it be fair to interpret the meaning of what Apple has called "Aggregate Device" as a "Multi-Input Device"? If so, what is the point of adding both the microphone (technically an input device) and also the speakers (technically and output device) to such an Aggregate Device? For example, there is no sound coming from the speakers if I add built-in speakers to such an aggregate device and select is as the sound output device of the system.


Slight difference between the two. Multi-output is stacked. When outputting audio to a multi output it send audio to channel 1-2 of each device on the list. And only works on outputs.

This isn’t true for an aggregate. When using an aggregate device you need to specify which channels to input/output audio.

Hope that makes sense.

3.

**Comment by gatisnolv** [comment]
Thanks.
Regarding 2, "In the new Aggregate device you just created select the audio devices that you want to **output audio**", I think then this could say something like "use as inputs and outputs" or maybe just "aggregate" but this sounds a bit vague.
